src/tls.rs

```diff
@@ -95,10 +95,26 @@ mod encryption {
                             let mut root_store = RootCertStore::empty();
                             #[cfg(feature = "rustls-tls-native-roots")]
                             {
-                                let native_certs = rustls_native_certs::load_native_certs()?;
-                                let total_number = native_certs.len();
+                                let rustls_native_certs::CertificateResult {
+                                    certs, errors, ..
+                                } = rustls_native_certs::load_native_certs();
+
+                                if !errors.is_empty() {
+                                    log::warn!(
+                                        "native root CA certificate loading errors: {errors:?}"
+                                    );
+                                }
+
+                                // Not finding any native root CA certificates is not fatal if the
+                                // "rustls-tls-native-roots" feature is enabled.
+                                #[cfg(not(feature = "rustls-tls-native-roots"))]
```

Comment:
This is in a `#[cfg(feature = "rustls-tls-native-roots")]` block, so would never be compiled in

Reply:
Thanks, it should be "rustls-tls-webpki-roots".